php/savesettings.php

```diff
@@ -119,6 +119,14 @@ function validDomain($domain_name)
 					$extra .= "no-bogus-priv";
 				}
 
+				// Check if Conditional Forwarding is requested
+				if(isset($_POST["conditionalForwarding"]))
+				{
+					$addressArray = explode(".", $_POST["conditionalForwardingIP"]);
+					$reverseAddress = $addressArray[2].".".$addressArray[1].".".$addressArray[0].".in-addr.arpa";
+					$extra .= " conditional_forwarding ".$_POST["conditionalForwardingIP"]." ".$_POST["conditionalForwardingDomain"]." $reverseAddress";
```

Comment:
This is a security issue. Please validate the IP and domain before using it, like how it's done in line 74 and 172:
```php
if (!validIP($primaryIP))
{
    $error .= "Primary IP (".$primaryIP.") is invalid!<br>";
}
// ...
if(!validDomain($domain))
{
    $error .= "Top Domains/Ads entry ".$domain." is invalid!<br>";
}
```

PHP makes it easy to create security issues, so always remember to never trust the client's data. In the current code, someone could add a script in the domain or IP address field and have it executed.